src/core/utils/exception.cpp

```diff
@@ -1,12 +1,13 @@
 ///////////////////////////////////////////////////////////////////////////////
 // BSD 3-Clause License
 //
-// Copyright (C) 2018-2019, LAAS-CNRS
+// Copyright (C) 2018-2020, University of Edinburgh
 // Copyright note valid unless otherwise stated in individual files.
 // All rights reserved.
 ///////////////////////////////////////////////////////////////////////////////
 
 #include "crocoddyl/core/utils/exception.hpp"
+#include "crocoddyl/core/utils/to-string.hpp"
```

Comment:
I would put this into the header "exception.hpp"

Reply:
I always include header in the place that are needed. Is this against a good-coding practice? If so, could you point out to a refer for my own curiosity?

Reply:
It's not needed at all, just use the string stream:
```
throw_pretty("Cost item doesn't have the same control dimension (it should be " << nu_ << ")");
```

Reply:
If you don't add it to the header, you will have to add it to every single file that uses `std::to_string`, which is in every place where we use `throw_pretty`

Reply:
We don't need to include in each place with this methods, indeed, I didn't have to do it.

Modifying each throw_pretty is plausible too, but what happens if we change in future the method? I believe it's more flexible to use `std::to_string`

Reply:
I think you missed my point. My point was: in many files where we use `throw_pretty`, there is an implicit `std::to_string`. So each of those files needs an include to the header `to-string.hpp`. If you include this header in `exception.hpp`, it is automatically included already.

Reply:
I saw your point, however, the files are compiling in Ubuntu 16.04 without need of including 'to-string.hpp'. Are you sure that your statement is right?

In my understanding, it's not needed because we're linking a library that contains it.